core/dbt/adapters/base/impl.py

```diff
@@ -1298,6 +1301,62 @@ def render_column_constraint(cls, constraint: ColumnLevelConstraint) -> str:
         else:
             return ""
 
+    @available
+    def render_column_constraint_ddl(self, columns: Dict[str, Dict]) -> List:
+        rendered_column_constraints = []
+        all_constraints = []
+
+        unsupported_contraints = self.list_unsupported_constraints()
+
+        for k, v in columns.items():
+            column_ddl = [f"{v['name']} {v['data_type']}"]
+            for con in v["constraints"]:
+                all_constraints.append(con)
+                if con["type"] not in unsupported_contraints:
+                    constraint = self._parse_column_constraint(con)
+                    column_ddl.append(self.render_column_constraint(constraint))
+            rendered_column_constraints.append(" ".join(column_ddl))
+
+        self.process_constraint_warnings(constraints=all_constraints)
```

Comment:
This may just come down to personal preference, but I find this method a bit difficult to read / duplicative because its helpers are also iterating over constraints to extract their type / support before performing some sort of action (filtering, warning, etc). I think much of it could be simplified if we inlined more here. For example, 

```python
rendered_column_constraints = []

# maybe this should be a constant class attribute instead of a static method, ie self.CONSTRAINT_SUPPORT
constraint_type_to_support = self.constraint_support()
constraints_by_type = {cs: [] for cs in ConstraintSupport}

for k, v in columns.items():
    rendered_column_constraint = [f"{v['name']} {v['data_type']}"]
    for con in v["constraints"]:
        constraint = self._parse_column_constraint(con) # parsing the raw column constraint earlier so we have a ColumnConstraint object to work with
        constraints_by_type[constraint.type].append(constraint)
        if constraint_type_to_support[con.type] != ConstraintSupport.NOT_SUPPORTED:
           rendered_column_constraint.append(self.render_column_constraint(constraint))
    rendered_column_constraints.append(" ".join(rendered_column_constraint))

self.raise_constraint_warnings(constraints_by_type) # this probably ends up being pretty simple, could consider inlining it as well. 
return rendered_column_constraints
```

Reply:
I took some of this into account.  But I discovered we can't bundle the error messages per table since the warn bools can be set per constraint.